### ISSUE TITLE
table_privs_are is falsely reporting extra privileges

### ISSUE THREAD

**Issue by machineghost** [theory/pgtap#303]
I have a test to confirm that a "public_anonymous" user can only SELECT from a particular table:

```
SELECT table_privs_are(
    'public', 'resources', 'public_anonymous', ARRAY['SELECT']
);

```

It is failing:
```
# Failed test 4: "Role public_anonymous should be granted SELECT on table public.resources"
#     Extra privileges:
#         INSERT
```

However, that role does not have INSERT privileges.  What's weird is I can even explicitly remove them:

```
# REVOKE INSERT ON resources FROM public_anonymous;
REVOKE
```

and I can see in DBeaver that the only permission that public_anonymous has on that table is `SELECT` 

![permissions](https://user-images.githubusercontent.com/448908/214470915-3ca3e04d-6d96-46f0-95a0-8c1b1e83265d.png)

... but the test still fails.

In other words, I'm pretty sure (though I could be wrong) that the user being tested *does not* have INSERT permissions ... so I don't understand why the test thinks they do.

**Comment by machineghost** [comment]
Here's the result of `\z resources;`.  As you can see, user public_anonymous has:

    public_anonymous=r/admin

ie. read (SELECT) and admin (whatever that is).  There is no "a", which (per https://www.postgresql.org/docs/12/ddl-priv.html#PRIVILEGE-ABBREVS-TABLE) is what it would say if that user had INSERT permissions.

Full output:

```
Schema |   Name    | Type  |       Access privileges        | Column privileges |                                                                                                Policies                                       
                                                          
--------+-----------+-------+--------------------------------+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------
 public | resources | table | admin=arwdDxt/admin           +|                   | select_resources (r):                                                                                                                         
                                                         +
        |           |       | public_postgraphile=arwd/admin+|                   |   (u): true                                                                                                                                   
                                                         +
        |           |       | public_anonymous=r/admin      +|                   | update_resources (w):                                                                                                                         
                                                         +
        |           |       | public_admin_user=arwd/admin  +|                   |   (u): (( SELECT resource_authors.user_id                                                                                                     
                                                         +
        |           |       | public_user=ar/admin          +|                   |    FROM resource_authors                                                                                                                      
                                                         +
        |           |       | =a/admin                       |                   |   WHERE ((resource_authors.resource_id = resource_authors.resource_id) AND (resource_authors.user_id = (NULLIF(current_setting('jwt.claims.person_id'::text, true), ''::text))::integer))) IS NOT NULL)+
        |           |       |                                |                   |   to: public_user                                                                                                                             
                                                         +
        |           |       |                                |                   | delete_resources (d):                                                                                                                         
                                                         +
        |           |       |                                |                   |   (u): (( SELECT resource_authors.user_id                                                                                                     
                                                         +
        |           |       |                                |                   |    FROM resource_authors                                                                                                                      
                                                         +
        |           |       |                                |                   |   WHERE ((resource_authors.resource_id = resource_authors.resource_id) AND (resource_authors.user_id = (NULLIF(current_setting('jwt.claims.per
son_id'::text, true), ''::text))::integer))) IS NOT NULL)+
        |           |       |                                |                   |   to: public_user
```

**Comment by theory** [comment]
I created a user named `public_anonymous`, and granted them SELECT access to table `foo`. Here's what I see:

```
david=# \z foo
                                Access privileges
 Schema | Name | Type  |    Access privileges     | Column privileges | Policies 
--------+------+-------+--------------------------+-------------------+----------
 public | foo  | table | david=arwdDxt/david     +|                   | 
        |      |       | public_anonymous=r/david |                   | 
```

So maybe the `/david` bit identifiers the role that granted the permission? Anyway, pgTAP relies on [a function named `_get_table_privs`](https://github.com/theory/pgtap/blob/c86203e/sql/pgtap.sql.in#L9079-L9102) to determine the privileges on a table. Assuming you have pgTAP installed in your database and it's in your path, you can just run it to see what it outputs. Here's what I see:

```
david=# select _get_table_privs('public_anonymous', 'foo');
 _get_table_privs 
------------------
 {SELECT}
```

Presumably yours will also show `INSERT`. I wonder if something in the policies column of your output suggests additional access? 

`_get_table_privs` in turn relies on the Postgres catalog function `pg_catalog.has_table_privilege()` to learn about privileges. For example:

```
david=# select pg_catalog.has_table_privilege('public_anonymous', 'foo', 'INSERT');
 has_table_privilege 
---------------------
 f
```

What is the output you get for this function (pass the schema-qualified table name as the second argument if it's not in your search path)?

**Comment by theory** [comment]
Hey, nice example of pgTAP helping you identify and fix an access control issue, yay! 🎉
